src/plots/mapbox/mapbox.js

```diff
@@ -172,11 +173,11 @@ proto.updateMap = function(calcData, fullLayout, resolve, reject) {
 
     self.rejectOnError(reject);
 
-    var styleUrl = convertStyleUrl(self.opts.style);
+    var styleObj = getStyleObj(self.opts.style);
 
-    if(self.styleUrl !== styleUrl) {
-        self.styleUrl = styleUrl;
-        map.setStyle(styleUrl);
+    if(self.styleObj.id !== styleObj.id) {
```

Comment:
cc @plotly/frontend simply mutating a mapbox style JSON won't trigger an update. Mapbox style JSONs are assumed to be constant unless their `id` change.